src/devices/cpu/ht1130/ht1130.cpp

```diff
@@ -0,0 +1,1049 @@
+
+		if (getacc())
+			m_pc = fulladdr;
+
+		return;
+	}
+
+	// case 0b10100aaa: (with 8-bit immediate) : JNZ R0,address : Jump if register is not 0
+	case 0b10100000: case 0b10100001: case 0b10100010: case 0b10100011:
+	case 0b10100100: case 0b10100101: case 0b10100110: case 0b10100111:
+	{
+		const u8 operand = fetch();
+		const uint16_t fulladdr = ((inst & 0x07) << 8) | operand | (m_pc & 0x800);
+
+		if (getreg(0))
+			m_pc = fulladdr;
+
+		return;
+	}
+
+	// case 0b10101aaa: (with 8-bit immediate) : JNZ R1,address : Jump if register is not 0
+	case 0b10101000: case 0b10101001: case 0b10101010: case 0b10101011:
+	case 0b10101100: case 0b10101101: case 0b10101110: case 0b10101111:
+	{
+		const u8 operand = fetch();
+		const uint16_t fulladdr = ((inst & 0x07) << 8) | operand | (m_pc & 0x800);
+
+		if (getreg(1))
+			m_pc = fulladdr;
+
+		return;
+	}
+
+	// case 0b11011aaa: (with 8-bit immediate) : JNZ R4,address : Jump if register is not 0
+	case 0b11011000: case 0b11011001: case 0b11011010: case 0b11011011:
+	case 0b11011100: case 0b11011101: case 0b11011110: case 0b11011111:
+	{
+		const u8 operand = fetch();
+		const uint16_t fulladdr = ((inst & 0x07) << 8) | operand | (m_pc & 0x800);
+
+		if (getreg(4))
+			m_pc = fulladdr;
+
+		return;
+	}
+
+	// case 0b11010aaa: (with 8-bit immediate) : JTMR address : Jump if time-out
+	case 0b11010000: case 0b11010001: case 0b11010010: case 0b11010011:
+	case 0b11010100: case 0b11010101: case 0b11010110: case 0b11010111:
+	{
+		const u8 operand = fetch();
+		const uint16_t fulladdr = ((inst & 0x07) << 8) | operand | (m_pc & 0x800);
+
+		if (m_timerover)
+		{
+			m_pc = fulladdr;
+			m_timerover = 0;
+		}
+		return;
+	}
+
+	// case 0b10110aaa: (with 8-bit immediate) : JZ A,address : Jump if accumulator is 0
+	case 0b10110000: case 0b10110001: case 0b10110010: case 0b10110011:
+	case 0b10110100: case 0b10110101: case 0b10110110: case 0b10110111:
+	{
+		const u8 operand = fetch();
+		const uint16_t fulladdr = ((inst & 0x07) << 8) | operand | (m_pc & 0x800);
+
+		if (!getacc())
+			m_pc = fulladdr;
+
+		return;
+	}
+
+	//case 0b100nnaaa: // (with 8-bit immediate) : JAn address : Jump if accumulator bit n is set
+	case 0b10000000: case 0b10000001: case 0b10000010: case 0b10000011:
+	case 0b10000100: case 0b10000101: case 0b10000110: case 0b10000111:
+	case 0b10001000: case 0b10001001: case 0b10001010: case 0b10001011:
+	case 0b10001100: case 0b10001101: case 0b10001110: case 0b10001111:
+	case 0b10010000: case 0b10010001: case 0b10010010: case 0b10010011:
+	case 0b10010100: case 0b10010101: case 0b10010110: case 0b10010111:
+	case 0b10011000: case 0b10011001: case 0b10011010: case 0b10011011:
+	case 0b10011100: case 0b10011101: case 0b10011110: case 0b10011111:
+	{
+		const u8 operand = fetch();
+		const uint16_t fulladdr = ((inst & 0x07) << 8) | operand | (m_pc & 0x800);
+		const uint8_t bit = (inst & 0x18) >> 3;
+
+		if (getacc() & (1<<bit))
```

Comment:
BIT(getacc(), bit)

Reply:
I still think these macros only serve to obfuscate, but done

Reply:
I only use it in these kind of cases (x & 1 << y) or (x >> y & 1).